src/settings/SettingsPage.js

```diff
@@ -2,23 +2,40 @@ import React from 'react'
 import { Helmet } from 'react-helmet'
 import { connect } from 'redux-bundler-react'
 import { translate, Trans } from 'react-i18next'
+import { getCurrentLanguage } from '../lib/i18n'
+
+// Components
+import Tick from '../icons/GlyphSmallTick'
 import Box from '../components/box/Box'
 import Button from '../components/button/Button'
 import JsonEditor from './editor/JsonEditor'
-import Tick from '../icons/GlyphSmallTick'
+import Overlay from '../components/overlay/Overlay'
+import LanguageModal from './language-modal/LanguageModal'
 
 const PAUSE_AFTER_SAVE_MS = 3000
 
 const SettingsPage = ({
   t, tReady,
   isConfigBlocked, isLoading, isSaving,
   hasSaveFailed, hasSaveSucceded, hasErrors, hasLocalChanges, hasExternalChanges,
-  config, onChange, onReset, onSave, editorKey
+  config, onChange, onReset, onSave, editorKey, showEditModal, onLanguageEditOpen, onLanguageEditClose
 }) => (
   <div data-id='SettingsPage'>
     <Helmet>
       <title>{t('title')} - IPFS</title>
     </Helmet>
+
+    <Box className='mb3 pa4'>
+      <div className='flex justify-between items-center'>
+        <p className='ma0 lh-copy charcoal-muted f6'>
+          {t('language', { language: getCurrentLanguage() })}
```

Comment:
UI suggestion, make the label look like our block labels, make the language name larger and more obvious... if you need to change it you're gonna need it big and clear, and bring the button next to it.

<img width="882" alt="screenshot 2018-10-31 at 11 06 03" src="https://user-images.githubusercontent.com/58871/47784330-23303a80-dcfd-11e8-9a5d-4371b276f332.png">

It could also be it's own compoent, that opens and closes the model, so the parent page doesn't need to know abou it.